skimage/io/tests/test_pil.py

```diff
@@ -45,18 +45,19 @@ def test_png_round_trip():
     fname = f.name
     f.close()
     I = np.eye(3)
-    imsave(fname, I)
+    with expected_warnings(['Possible precision loss']):
+        imsave(fname, I)
     Ip = img_as_float(imread(fname))
     os.remove(fname)
     assert np.sum(np.abs(Ip-I)) < 1e-3
 
 
 def test_imread_flatten():
     # a color image is flattened
-    img = imread(os.path.join(data_dir, 'color.png'), flatten=True)
+    img = imread(os.path.join(data_dir, 'color.png'), as_gray=True)
```

Comment:
If I'm not mistaken, this test should remain until `flatten` is removed, but with `expected_warnings`.

Reply:
Ok, I added this test:
```python
def test_img_as_grey_flatten():
    img = imread(os.path.join(data_dir, 'color.png'), as_gray=True)
    with expected_warnings(['deprecated']):
        img_flat = imread(os.path.join(data_dir, 'color.png'), flatten=True)
    assert_array_equal(img, img_flat)
```